### PR TITLE
Add Active Directory LDAPS deploy

### PR DESCRIPTION
**Pull request by DACRepair**
This adds `Set-ActiveDirectoryLDAPS` to the module. Since AD is opinionated (IE domain has to be correct), but the deploy requires that a set of common params be sent: I am leaving it up to the user to get right.

Usage is the same as most of the other modules, however in order to get the correct cert I would recommend this:
```powershell
# Create the certificate
$dc = Get-ADDomainController -Discover
$domains = $dc.HostName, $dc.Domain

New-PACertificate -Domain $domains | Set-ActiveDirectoryLDAPS -Verbose -RemoveOldCert
```

### CODE REVIEW

**Comment by rmbolger** [comment]
Ooh, fun! Thanks @DACRepair. Let me play around with this a bit and we'll get it merged.

**Comment by rmbolger** [comment]
Ok, I'm finally getting around to messing with this. So sorry for the delay. My only functional question is around the `-RemoveOldCert` flag. Is there a reason we're only removing the old cert from the NTDS store and not the Local Computer store where the cert also gets imported? I feel like people will expect both to get cleaned up.

I'm going to add a doc page for completeness, not that there's much to document with this one.

**Comment by rmbolger** [comment]
I updated the RemoveOldCert code to cleanup both store locations. But now that I'm thinking about it again, I can see how perhaps deleting the LocalMachine store copy wouldn't necessarily be desirable if you're using that copy for other things as well.

It seems like from a AD perspective, you'd always want only 1 cert in the NTDS store so it has no choice about what to use. So maybe we always delete the old in NTDS and then use `-RemoveOldCert` only to decide about the LocalMachine store? What do you think, @DACRepair?

**Comment by DACRepair** [comment]
Hey I just wanted to say I appreciate the updates. I needed to revisit this as it has been a little while since working on it. This flag was simply to keep the parameter pattern the other plugins were using.

```
Is there a reason we're only removing the old cert from the NTDS store and not the Local Computer store where the cert also gets imported? I feel like people will expect both to get cleaned up.
```
This is simply an oversight on my part. You are correct with this.

```
It seems like from a AD perspective, you'd always want only 1 cert in the NTDS store so it has no choice about what to use. So maybe we always delete the old in NTDS and then use -RemoveOldCert only to decide about the LocalMachine store?
```
Deleting both is ok. _Technically_ if the NTDS store cert is missing: LDAPS will turn off, but AD will continue other functions as normal. I agree with your previous above statement and feel having both artifacts removed when the flag is set would be appropriate. 

In short: I'd stick to the pattern, I cannot think of any "AD" reason not to do so.

**Comment by rmbolger** [comment]
With the current changes I made, it will delete both copies of the old cert but only if you specify the `-RemoveOldCert` switch. Conversely, if the switch is not used, the NTDS store will generally end up with 2 valid certs to choose from at least for the 30'ish days the old one remains unexpired given the default renewal windows. But the whole point of tweaking the NTDS store directly is to take away that ambiguity.

So I'm thinking that it makes more logical sense to *always* remove the old cert in the NTDS store and only make the switch responsible for the copy in LocalMachine. That's more like how it works for things like IIS. Not using the switch leaves old certs still on the system, but the NTDS service is only ever assigned the newest one.

**Comment by rmbolger** [comment]
Ok, made that change and fixed an embarrassing bug I added from before. We're good to go.
